### PR TITLE
[Merged by Bors] - refactor(Tactic/ToAdditive): avoid FindM, got removed in lean4

### PR DESCRIPTION
**Pull request by nomeata**
this code was using an internal data type from Lean4 that was removed in https://github.com/leanprover/lean4/pull/4795 so let's inline the three lines that we need here.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/ToAdditive/Frontend.lean [hunk 1]
```diff
@@ -486,13 +485,15 @@ open Lean.Expr.FindImpl in
   and we're not remembering the cache between these calls. -/
 unsafe def additiveTestUnsafe (findTranslation? : Name → Option Name)
   (ignore : Name → Option (List ℕ)) (e : Expr) : Option Name :=
-  let rec visit (e : Expr) (inApp := false) : OptionT FindM Name := do
+  let rec visit (e : Expr) (inApp := false) : OptionT (StateM (PtrSet Expr)) Name := do
```
Comment (grunweg) [inline Mathlib/Tactic/ToAdditive/Frontend.lean:488]: I'm just have an extremely local understanding of the code, so take this with a grain of salt: this was `StateT` in Lean core; was this change intentional?

**Review by grunweg — COMMENTED** [review]
One question; the rest *looks* good to me. (But note that I'm just locally verifying things; I have no global understanding whatsoever of this code.)

**Comment by grunweg** [comment]
Also, let me add: thanks for taking care of mathlib this way!

File: Mathlib/Tactic/ToAdditive/Frontend.lean (hunk 1 above)
Comment (nomeata) [reply Mathlib/Tactic/ToAdditive/Frontend.lean:488]: Thanks for asking. It was `StateT (PtrSet Expr) Id`, which is the same as `StateM (PtrSet Expr)`

Comment (grunweg) [reply Mathlib/Tactic/ToAdditive/Frontend.lean:488]: :bulb: Makes sense, thanks!

**Review by grunweg — COMMENTED** [review]
According to my (necessarily shallow) review, this looks good.
maintainer merge

**Comment by semorrison** [comment]
bors merge
